Dalamud/Game/Config/GameConfig.cs

```diff
@@ -192,3 +188,178 @@ private unsafe nint OnConfigChanged(ConfigBase* configBase, ConfigEntry* configE
         return returnValue;
     }
 }
+
+/// <summary>
+/// Plugin-scoped version of a GameConfig service.
+/// </summary>
+[PluginInterface]
+[InterfaceVersion("1.0")]
+[ServiceManager.ScopedService]
+#pragma warning disable SA1015
+[ResolveVia<IGameConfig>]
+#pragma warning restore SA1015
+internal class GameConfigPluginScoped : IDisposable, IServiceType, IGameConfig
+{
+    [ServiceManager.ServiceDependency]
+    private readonly GameConfig gameConfigService = Service<GameConfig>.Get();
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="GameConfigPluginScoped"/> class.
+    /// </summary>
+    internal GameConfigPluginScoped()
+    {
+        this.gameConfigService.Changed += this.ConfigChangedForward;
+    }
+    
+    /// <inheritdoc/>
+    public event EventHandler<ConfigChangeEvent>? Changed;
+    
+    /// <inheritdoc/>
+    public GameConfigSection System => this.gameConfigService.System;
```

Comment:
Individual `GameConfigSection`s have their own public `Changed` event. Unsure how to fix this, but it does cause issues with scoping. Will defer to Goat/Cara.

Reply:
One avenue might be to make the GameConfigSection.Changed event internal, and add those events to the interface then have pluginscoped act as a proxy for invoking the event internally.

Reply:
Prototype of potential solution: https://github.com/MidoriKami/Dalamud/commit/4a2d173925ef5c1481ceb691f99030a01cd782d7